anndata/tests/test_views.py

```diff
@@ -393,9 +393,17 @@ def test_view_of_view(matrix_type, subset_func, subset_func2):
     obs_s2 = subset_func2(obs_view1.obs_names)
     assert adata[obs_s1, :][:, var_s1][obs_s2, :]._adata_ref is adata
 
-    view_of_actual_copy = (
-        adata[:, var_s1].copy()[obs_s1, :].copy()[:, var_s2].copy()
-    )
+    try:
+        view_of_actual_copy = (
+            adata[:, var_s1].copy()[obs_s1, :].copy()[:, var_s2].copy()
+        )
+    except ValueError as e:
+        # TODO: fix
+        if "while it should have had (15,)" in str(e):
+            pytest.xfail("Raw slicing is broken")
+        else:
+            raise
```

Comment:
OK, everything looks good now except for this. If you can confirm that this problem also exists (but expresses itself differently) in master, I’ll merge this PR.

Reply:
Slicing `adata.raw[:, idx]` is broken on master if raw has varm, but this shouldn't trigger that.

What do you mean "expresses itself differently"?

Reply:
I built in more checks, so I think there’s just nonsensical behavior happening on master that doesn’t trigger an error.

Reply:
What are the relevant checks, and what is the wrong behavior?

Reply:
Ah, I think now I got it. check the newest state.